mailbox/api/src/main/java/org/apache/james/mailbox/MessageIdManager.java

```diff
@@ -39,4 +42,6 @@
     void delete(MessageId messageId, List<MailboxId> mailboxIds, MailboxSession mailboxSession) throws MailboxException;
 
     void setInMailboxes(MessageId messageId, List<MailboxId> mailboxIds, MailboxSession mailboxSession) throws MailboxException;
+
+    MessageMetaData appendMessage(InputStream msgIn, Date internalDate, Flags flagsToBeSet, MailboxId mailboxId, MailboxSession mailboxSession) throws MailboxException;
```

Comment:
@mbaechler : so caller uses both MessageManager AND MessageIdManager ? I'm OK, it means less work for me.

I will keep the refactoring introduced for this, as it is still valuable in my opinion.